Corona-Warn-App/src/main/java/de/rki/coronawarnapp/dccticketing/core/common/DccTicketingException.kt

```diff
@@ -24,86 +30,90 @@ class DccTicketingException(
-        VD_ID_NO_ATS("VD_ID_NO_ATS"),
-        VD_ID_NO_NETWORK("VD_ID_NO_NETWORK"),
-        VD_ID_NO_VS_SVC_KEY("VD_ID_NO_VS_SVC_KEY"),
-        VD_ID_NO_VS("VD_ID_NO_VS"),
-        VD_ID_PARSE_ERR("VD_ID_PARSE_ERR"),
-        VD_ID_SERVER_ERR("VD_ID_SERVER_ERR"),
-        VD_ID_EMPTY_X5C("VD_ID_EMPTY_X5C"),
+        VD_ID_CLIENT_ERR(message = VD_ID_ERROR_MSG),
+        VD_ID_NO_ATS_SIGN_KEY(message = VD_ID_ERROR_MSG),
+        VD_ID_NO_ATS_SVC_KEY(message = VD_ID_ERROR_MSG),
+        VD_ID_NO_ATS(message = VD_ID_ERROR_MSG),
+        VD_ID_NO_NETWORK(message = VD_ID_ERROR_MSG, textKey = TextKey.TRY_AGAIN),
+        VD_ID_NO_VS_SVC_KEY(message = VD_ID_ERROR_MSG),
+        VD_ID_NO_VS(message = VD_ID_ERROR_MSG),
+        VD_ID_PARSE_ERR(message = VD_ID_ERROR_MSG),
+        VD_ID_SERVER_ERR(message = VD_ID_ERROR_MSG, textKey = TextKey.TRY_AGAIN),
+        VD_ID_EMPTY_X5C(message = VD_ID_ERROR_MSG),
 
         /**
          * Validation Service error codes
          * Failure when obtaining Service Identity Document of Validation Service; user may retry
          */
-        VS_ID_CERT_PIN_MISMATCH("VS_ID_CERT_PIN_MISMATCH"),
-        VS_ID_CERT_PIN_NO_JWK_FOR_KID("VS_ID_CERT_PIN_NO_JWK_FOR_KID"),
-        VS_ID_CLIENT_ERR("VS_ID_CLIENT_ERR"),
-        VS_ID_NO_ENC_KEY("VS_ID_NO_ENC_KEY"),
-        VS_ID_NO_NETWORK("VS_ID_NO_NETWORK"),
-        VS_ID_NO_SIGN_KEY("VS_ID_NO_SIGN_KEY"),
-        VS_ID_PARSE_ERR("VS_ID_PARSE_ERR"),
-        VS_ID_SERVER_ERR("VS_ID_SERVER_ERR"),
-        VS_ID_EMPTY_X5C("VS_ID_EMPTY_X5C")
+        VS_ID_CERT_PIN_MISMATCH(message = VS_ID_ERROR_MSG),
+        VS_ID_CERT_PIN_NO_JWK_FOR_KID(message = VS_ID_ERROR_MSG),
+        VS_ID_CLIENT_ERR(message = VS_ID_ERROR_MSG),
+        VS_ID_NO_ENC_KEY(message = VS_ID_ERROR_MSG),
+        VS_ID_NO_NETWORK(message = VS_ID_ERROR_MSG, textKey = TextKey.TRY_AGAIN),
+        VS_ID_NO_SIGN_KEY(message = VS_ID_ERROR_MSG),
+        VS_ID_PARSE_ERR(message = VS_ID_ERROR_MSG),
+        VS_ID_SERVER_ERR(message = VS_ID_ERROR_MSG, textKey = TextKey.TRY_AGAIN),
+        VS_ID_EMPTY_X5C(message = VS_ID_ERROR_MSG)
     }
 
-    // to-do: Add all error codes
-    val errorMessage: LazyString
-        get() = CachedString { context ->
-            when (errorCode) {
-                ErrorCode.VS_ID_SERVER_ERR -> R.string.dcc_ticketing_error_try_again
-                else -> ERROR_MESSAGE_GENERIC
-            }.let { context.getString(it) }
-        }
+    private fun ErrorCode.errorMessageRes(serviceProvider: String): LazyString = when (textKey) {
+        TextKey.SERVICE_PROVIDER_ERROR_NO_NAME ->
+            R.string.dcc_ticketing_error_service_provider_error_no_name.toResolvingString()
+        TextKey.SERVICE_PROVIDER_ERROR ->
+            R.string.dcc_ticketing_error_service_provider_error.toResolvingString(
+                serviceProvider
+            )
+        TextKey.TRY_AGAIN ->
+            R.string.dcc_ticketing_error_try_again.toResolvingString()
+    }
```

Comment:
@jurajkusnier  is this something you already doing?

Reply:
yes, I put this my task as obsolete as this solution looks good enough